ui/src/admin/containers/DatabaseManagerPage.js

```diff
@@ -84,6 +87,10 @@ class DatabaseManagerPage extends Component {
         return notify('error', 'Database name cannot be blank')
       }
 
+      if (_.findIndex(databases, {name: database.name}, 1) !== -1) {
+        return notify('error', 'A database by this name already exists')
+      }
+
```

Comment:
The need for duplication here makes me think we can make a small improvement. Since from L86 to L94, there's no references to `e` defined on L78, we can probably replace that entire block with a call to `handleCreateDatabase(database)`

Reply:
There is a reference to `key` just cant see it in the diff.  This block is necessary to protect against an optimistic update of state on `Enter`.

Reply:
Ah, okay. Disregard then